Pagewizard & ProgressIndicator update

rawa-resul opened the pull request:
Closes #1094 #476

**Summary**

Added missing `secondaryLabel` prop to PageWizard's ProgressIndicator and fixed some styling for the ProgressIndicator in general.

**Change List (commits, features, bugs, etc)**

_PageWizard_

- Added `secondaryLabel` prop to `ProgressIndicator`
- Used native vertical prop instead of injected className for `ProgressIndicator`

_PageWizard.story_

- Added `secondaryLabel` prop to `ProgressIndicator`

_PageWizard.story.storyshot_

- Updated snapshot to reflect changes

_ProgressIndicator_

- Replaced all hard-coded `bx` with `${prefix}`
- Used native vertical prop instead of injected className

_ProgressIndicator.story_

- Changed the value "secondary label" to "Optional label" to make it clearer
- hideLabels and default stepWidth example never hid the labels - added prop to hide them and optional vertical mode knob

_ProgressIndicator.story.storyshot_

- Updated snapshot to reflect changes

_ProgressIndicator CSS_

- Fixed issue whereby `progress-line` wouldn’t adjust to custom `stepWidth` #476
- Fixed broken vertical mode in `rtl`
- Changed positioning of `step button` elements back to prior to carbon upgrade. Aligned at top rather than centre.
- Generally, making `ltr` and `rtl` perfectly mirrored. 

**Acceptance Test (how to verify the PR)**

_PageWizard_
- Go to the PageWizard story
- Click on stateful example
- You should see optional label displayed and aligned with step label
- Click on RTL in storybook addon
- You should see ProgressIndicator perfectly mirrored and optional label aligned with step label

_ProgressIndicator_
- Go to ProgressIndicator
- Click on Presentation example
- You should see a contionous progress-line and no breaks in between
- Confirm by playing around with the stepWidth number in the storybook addon
- Click on hideLabels and default stepWidth example
- You should now only see the current step label displayed

bjornalm commented on src/components/ProgressIndicator/_progress-indicator.scss:
```diff
@@ -2,61 +2,82 @@
-        width: 1rem;
-      }
+    .#{$prefix}--progress-step svg {
+      margin-left: 0.563rem;
```
You could perhaps put this in a sass var and name it since it is used in more than one places?

bjornalm commented on src/components/ProgressIndicator/_progress-indicator.scss:
```diff
@@ -2,61 +2,82 @@
+      }
+
+      .#{$prefix}--progress-optional {
+        margin-right: ($spacing-06 + 1.1rem);
```
this calculation is also repeated, consider sass var?

bjornalm commented:
Did the spec say anything about text overflow?
![image](https://user-images.githubusercontent.com/5167091/80214074-7a854180-863a-11ea-8f60-ffc76892b59c.png)

tay1orjones commented:
> Did the spec say anything about text overflow?
> ![image](https://user-images.githubusercontent.com/5167091/80214074-7a854180-863a-11ea-8f60-ffc76892b59c.png)

This defect is present in the Carbon version as well
![image](https://user-images.githubusercontent.com/3360588/80258074-f3d76100-8647-11ea-8355-927a316c3bb5.png)

rawa-resul commented:
@bjornalm Thanks for your review. I have removed all numbered values and replaced them with carbon spacing variables and some clean up. 
As for the overflow bug I tried a few things and nothing really worked as desired. Also, I think that is beyong the purpose of this PR. For the vertical state I added an overflow:hidden for now. I will create a separate issue for it.

tay1orjones commented:
[Internal Watson-IoT/pal-tracking#389](https://github.ibm.com/Watson-IoT/pal-tracking/issues/389)

tay1orjones commented:
:tada: This PR is included in version 2.72.2 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/carbon-addons-iot-react)
- [GitHub release](https://github.com/IBM/carbon-addons-iot-react/releases/tag/v2.72.2)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
